Metadata/MetadataConverter.php

```diff
@@ -36,43 +37,55 @@ public function convert(ClassHierarchyMetadata $metadata)
         static $count = 0;
         $definitions = array();
 
+        /** @var ClassMetadata $previous */
         $previous = null;
+        /** @var ClassMetadata $classMetadata */
         foreach ($metadata->classMetadata as $classMetadata) {
-            if (null === $previous && null === $classMetadata->parent) {
-                $definition = new Definition();
-            } else {
-                $definition = new DefinitionDecorator(
-                    $classMetadata->parent ?: $previous->id
-                );
-            }
+            foreach ($classMetadata->getServices() as $service) {
+                if (null === $previous && empty($service['parent'])) {
+                    $definition = new Definition();
+                } else {
+                    if (empty($service['parent']) && sizeof($previous->getServices()) > 1) {
```

Comment:
I can change it to !isset() - else I have to readd the @ (or copy&paste the null values for every addService call)